ports/glutin/window.rs

```diff
@@ -470,81 +446,158 @@ impl Window {
+                event: glutin::WindowEvent::ReceivedCharacter(ch),
+            } => self.handle_received_character(ch),
+            Event::WindowEvent {
+                window_id: _,
+                event: glutin::WindowEvent::KeyboardInput {
+                    device_id: _,
+                    input: glutin::KeyboardInput {
+                        scancode: scancode,
+                        state: element_state,
+                        virtual_keycode: Some(virtual_keycode),
+                        modifiers: _,
+                    },
+                },
+            } => {
+                self.handle_keyboard_input(element_state, scancode, virtual_keycode);
+            },
+            Event::WindowEvent {
+                window_id: _,
+                event: glutin::WindowEvent::KeyboardInput {
+                    device_id: _,
+                    input: glutin::KeyboardInput {
+                        scancode: _,
+                        state: _,
+                        virtual_keycode: None,
+                        modifiers: _,
+                    },
+                },
+            } => {
                 debug!("Keyboard input without virtual key.");
-            }
-            Event::Resized(width, height) => {
-                self.event_queue.borrow_mut().push(WindowEvent::Resize(TypedSize2D::new(width, height)));
-            }
-            Event::MouseInput(element_state, mouse_button, pos) => {
-                if mouse_button == MouseButton::Left ||
-                   mouse_button == MouseButton::Right {
-                    match pos {
-                        Some((x, y)) => {
-                            self.mouse_pos.set(Point2D::new(x, y));
-                            self.event_queue.borrow_mut().push(
-                                WindowEvent::MouseWindowMoveEventClass(TypedPoint2D::new(x as f32, y as f32)));
-                            self.handle_mouse(mouse_button, element_state, x, y);
-                        }
-                        None => {
-                            let mouse_pos = self.mouse_pos.get();
-                            self.handle_mouse(mouse_button, element_state, mouse_pos.x, mouse_pos.y);
-                        }
-                    }
+            },
+            Event::WindowEvent {
+                window_id: _,
+                event: glutin::WindowEvent::Resized(width, height),
+            } => {
+                self.event_queue.borrow_mut().push(WindowEvent::Resize(
+                    TypedSize2D::new(width, height),
+                ));
+            },
+            Event::WindowEvent {
+                window_id: _,
+                event: glutin::WindowEvent::MouseInput {
+                    device_id: _,
+                    state: state,
+                    button: button,
+                },
+            } => {
+                // TODO: position is necessary according to @paulrouget, but pos is not given in
+                // the event's struct.
```

Comment:
Yes. The PR that brings these coordinates to the wheel and click events didn't make it to upstream iirc.